dynamo/preprocessing/Preprocessor.py

```diff
@@ -445,8 +461,9 @@ def config_monocle_recipe(
 
         # recipe monocle log1p all raw data in normalize_by_cells (dynamo version), so we do not need extra log1p transform.
         self.use_log1p = False
+        self.regress_out_kwargs = {"variables": regress_out_obs_keys, "selected_genes": "use_for_pca"}
         self.pca = pca_monocle
-        self.pca_kwargs = {"pca_key": "X_pca"}
+        self.pca_kwargs = {"pca_key": "X_pca", "layer": "residuals_for_pca"}
```

Comment:
I have created a key in the 'obsm' dictionary that contains information about the residuals after performing the regressing out operation. This layer is necessary when the 'selected_genes' option is set to 'use_for_pca' for the regressing out process.

`selected_genes: the key in adata.var that contains boolean for showing genes` filtering results. For example, "use_for_pca" is selected then it will regress out only for the genes that are True for "use_for_pca". This input will decrease processing time of regressing out data.`

Since regressing out can take a long time, I believe that this option and the 'residuals_for_pca' layer are necessary for fast and efficient analysis. If you have any suggestions or better ideas, I would be grateful if you could share them with me.

Reply:
Updated as "layer": "regress_out"